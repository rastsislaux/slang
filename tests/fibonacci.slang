/*
    THIS FILE IS A PART OF SLANG EXAMPLES.
    AUTHOR: Rostislav Lipsky (https://github.com/ungaf)
    DATE: 06.02.2023
    VISIT https://github.com/ungaf/slang AND READ THE LICENSE BEFORE USING.

    This program prints fibonacci sequence from 2 to 144.
*/

include std.io

const CAP 0x90

macro main using std.io namespaces

    1 1
    cycle:
        dup rot +
        dup putx!
        " (" pop! dup puti! ")" pop!
        dup CAP! >
        if
            ", " pop!
            cycle~
        fi
    drop drop
    "\n" pop!
    
end