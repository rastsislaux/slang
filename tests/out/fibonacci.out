0x2 (2), 0x3 (3), 0x5 (5), 0x8 (8), 0xd (13), 0x15 (21), 0x22 (34), 0x37 (55), 0x59 (89), 0x90 (144)
